File: Quinta/radice_quadrata/Erone/radice1.cs
using System;

namespace csharp_test
{
    class Program
    {
        const int MAX = 10;

        static double radice_q(double num)
        {
            double Q = 0, QPrec;
            int n = 0;
            QPrec = num / 2;
            for (n=0; n<MAX; n++){
                Q = (QPrec + num / QPrec) / 2;
                QPrec = Q;
            }
            return Q;
        }

        static void Main(string[] args)
        {
            double numero;
            
            Console.WriteLine("Calcolo approssimato radice\n");
            Console.Write("inserisci un valore positivo :");

            numero = Convert.ToDouble(Console.ReadLine());

            Console.WriteLine("Approssimato:" + radice_q(numero));
            Console.WriteLine("Preciso     :" + Math.Sqrt(numero));
        }
    }
} 

File: Quinta/radice_quadrata/Erone/Radice1.java
package radice_quadrata.Erone;
import java.util.Scanner;
public class Radice1 {
    public static int MAX = 10;

    static double radice_q(double num){
        double Q = 0, QPrec;
        int n = 0;
        QPrec = num / 2;
        for (n=0; n<MAX; n++){
            Q = (QPrec + num / QPrec) / 2;
            QPrec = Q;
        }
        return Q;
    }

public static void main(String []arg){
    Scanner in = new Scanner(System.in);
    double numero;
    System.out.println("Calcolo approssimato radice \n");
    System.out.print("inserisci un valore positivo: ");
    numero = in.nextInt();
    System.out.println("Approssimato:" + radice_q(numero));
    System.out.println("Preciso     :" + Math.sqrt(numero));
}
}